verticapy/tests_new/machine_learning/vertica/test_base_model_methods.py

```diff
@@ -1343,144 +1115,244 @@ def test_get_vertica_attributes(
+        **dict.fromkeys(CLASSIFICATION_MODELS, "survived_pred"),
+        **dict.fromkeys(TIMESERIES_MODELS, "prediction"),
+        **dict.fromkeys(["KMeans"], f"{model_class}_cluster_ids"),
+    }
+    return pred_col_map.get(model_class, None)
 
-    def test_summarize(self, get_models):
-        """
-        test function - summarize
-        """
-        vpy_model_summary = get_models.vpy.model.summarize()
-        vt_model_summary_sql = f"SELECT GET_MODEL_SUMMARY(USING PARAMETERS MODEL_NAME='{get_models.vpy.schema_name}.{get_models.vpy.model_name}')"
-        vt_model_summary = (
-            current_cursor().execute(vt_model_summary_sql).fetchall()[0][0]
-        )
 
-        assert vpy_model_summary == vt_model_summary
+@pytest.fixture(name="get_to_sql")
+def get_to_sql_fixture(model_class, get_models):
+    """
+    getter fixture to get sql function for a model
+    """
+    if model_class in TIMESERIES_MODELS:
+        pytest.skip(f"to_sql function is not available for {model_class} model")
+
+    pred_func = get_predict_function(model_class)
+    model_name = get_models.vpy.model.model_name
+    to_sql_func = get_models.vpy.model.to_sql
+
+    to_sql_map = {
+        **dict.fromkeys(
+            REGRESSION_MODELS,
+            f"SELECT {pred_func}(3.0, 11.0, 93.0 USING PARAMETERS model_name = '{model_name}', match_by_pos=True)::float, {to_sql_func([3.0, 11.0, 93.0])}::float"
+            if model_class in REGRESSION_MODELS
+            else None,
+        ),
+        **dict.fromkeys(
+            CLASSIFICATION_MODELS,
+            f"""SELECT {pred_func}(* USING PARAMETERS model_name = '{model_name}', match_by_pos=True)::int, {to_sql_func()}::int FROM (SELECT 30.0 AS age, 45.0 AS fare, 'male' AS sex) x"""
+            if model_class in CLASSIFICATION_MODELS
+            else None,
+        ),
+        **dict.fromkeys(
+            ["KMeans"],
+            f"SELECT {pred_func}(3.0, 11.0, 93.0, 0.244 USING PARAMETERS model_name = '{model_name}', match_by_pos=True)::float, {to_sql_func([3.0, 11.0, 93.0, 0.244])}::float"
+            if model_class == "KMeans"
+            else None,
+        ),
+    }
+    yield to_sql_map.get(model_class, None)
 
-    def test_to_python(
-        self,
-        get_models,
-        model_class,
-    ):
+
+def _get_deploysql(model_class):
+    """
+    test function - deploySQL
+    """
+    deploysql_map = {
+        **dict.fromkeys(
+            REGRESSION_MODELS + CLASSIFICATION_MODELS + CLUSTER_MODELS,
+            """{pred_fun_name}({columns} USING PARAMETERS model_name = '{schema_name}.{model_name}', match_by_pos = 'true')""",
+        ),
+        **dict.fromkeys(
+            TIMESERIES_MODELS,
+            """{pred_fun_name}( USING PARAMETERS model_name = '{schema_name}.{model_name}', add_mean = True, npredictions = 10 ) OVER ()""",
+        ),
+    }
+    return deploysql_map.get(model_class, None)
+
+
+pytestmark = pytest.mark.parametrize(
```

Comment:
@vikash018 where do you actually use this? I am intrigued.

Reply:
This declares pytest.mark.parametrize at global level which would be applied to all test functions/methods in this the module.

https://pytest.org/en/7.4.x/reference/reference.html#globalvar-pytestmark